Fix TypeError in select_graphic_rendition

mustitz opened the pull request:
Fixes #202

- Add `private` parameter to select_graphic_rendition() method signature
- Handle `private=True` parameter by returning early (no-op for private SGR sequences)

superbobry commented on pyte/screens.py:
```diff
@@ -971,11 +971,14 @@ def alignment_display(self) -> None:
                 self.buffer[y][x] = self.buffer[y][x]._replace(data="E")
 
-    def select_graphic_rendition(self, *attrs: int) -> None:
+    def select_graphic_rendition(self, *attrs: int, **kwargs: Any) -> None:
```
Can this be `private: bool = False` so that we don't allow arbitrary kwargs here?

mustitz replied:
Yes, you're absolutely right. Using `private: bool = False` is much better than `**kwargs`.

I used the existing `**kwargs` pattern as a reference from other methods in the codebase, but you're correct that explicit parameters are better. I noticed that `set_mode()`, `reset_mode()`, `erase_in_display()`, and `report_device_attributes()` also use the same pattern for the private parameter. Should I refactor those methods for consistency as well?

superbobry replied:
Let's keep the rest as is and only change `select_graphic_rendition`.

mustitz replied:
Done

superbobry reviewed (approved):
Thanks!

superbobry commented:
Would you mind adding a test for this to avoid regressions in the future?

mustitz committed:
Add test for private SGR sequence handling

- Add test_private_sgr_sequence_ignored() to verify Vim 9.0+ modifyOtherKeys query is ignored
- Test ensures private SGR sequences don't affect screen display, modes, or cursor state

mustitz committed:
Fix TypeError in select_graphic_rendition

- Add private parameter to select_graphic_rendition() method signature
- Handle private=True parameter by returning early (no-op for private SGR sequences)

mustitz commented:
> Would you mind adding a test for this to avoid regressions in the future?

Done
